pkgs/development/python-modules/deltalake/default.nix

```
@@ -0,0 +1,74 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pyarrow
+, pyarrow-hotfix
+, rustPlatform
+, perl
+, pytestCheckHook
+, pytest-benchmark
+, pandas
+}:
+
+buildPythonPackage rec {
+  pname = "deltalake";
+  version = "0.18.1";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "delta-io";
+    repo = "delta-rs";
+    rev = "refs/tags/python-v${version}";
+    hash = "sha256-R+fyxNbPhS04G7jWPopFNIWmZ4VW0SV0JKa4zBcV3Vg=";
+  };
+
+  postPatch = ''
+    # No Cargo.lock file in source
+    ln -s ${./Cargo.lock} Cargo.lock
+ '';
+
+ cargoDeps = rustPlatform.importCargoLock {
+   lockFile = ./Cargo.lock;
+ };
+
+  buildAndTestSubdir = "python"; # Make maturin find python package
+
+  preCheck = ''
+    # For paths in test to work, we have to be in python dir
+    cd python
+
+    # In tests we want to use deltalake that we have built
+    rm -rf deltalake
+    rm pyproject.toml
+  '';
+
+  dependencies = [
+    pyarrow
+    pyarrow-hotfix
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pandas
+    pytest-benchmark
+  ];
+
+  pytestFlagsArray = [ "-m 'not integration'" ];
+
+  nativeBuildInputs = [
+    perl # openssl-sys needs this
+  ] ++ (with rustPlatform; [
+    cargoSetupHook
+    maturinBuildHook
+  ]);
+
+  pythonImportsCheck = [ "deltalake" ];
+
+  meta = with lib; {
+    description = "A native Rust library for Delta Lake, with bindings into Python";
```

```suggestion
    description = "Native Rust library for Delta Lake, with bindings into Python";
```

https://github.com/NixOS/nixpkgs/pull/317959 😅